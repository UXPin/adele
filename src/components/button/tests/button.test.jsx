import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import Button from '../button';

const mockFunc = () => true;

it('renders filled Button with button markup correctly', () => {
  const wrapper = mount(<Button type="button" label="Start Exploring Systems" action={() => mockFunc} tab={1} />);
  expect(wrapper).toMatchSnapshot();
});
