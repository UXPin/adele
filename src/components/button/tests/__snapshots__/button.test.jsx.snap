// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders filled Button with button markup correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  height: 40px;
  min-height: 40px;
  width: 100%;
  max-width: 200px;
  background-color: #006cff;
  border-radius: 25px;
  border: 1px solid #006cff;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: 300;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: all ease-in 0.1s;
  transition: all ease-in 0.1s;
}

.c0:hover {
  background-color: #3389ff;
  border-color: #3389ff;
  box-shadow: 4px 4px 20px rgba(179,211,255,0.6);
}

.c0:active {
  background-color: #0061e6;
  box-shadow: 'none';
}

.c0:focus {
  outline: none;
  background-color: #3389ff;
}

<Button
  action={[Function]}
  label="Start Exploring Systems"
  tab={1}
  targetBlank={false}
  type="button"
>
  <styled.button
    onClick={[Function]}
    tabIndex={1}
  >
    <button
      className="c0"
      onClick={[Function]}
      tabIndex={1}
    >
      Start Exploring Systems
    </button>
  </styled.button>
</Button>
`;
